lib/postgres_test.go

```diff
@@ -1,10 +1,284 @@
 package lib
 
 import (
+	"context"
+	"fmt"
+	"github.com/deso-protocol/core/migrate"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/go-pg/pg/v10"
+	migrations "github.com/robinjoseph08/go-pg-migrations/v3"
 	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
 )
 
+// Set to true if you don't want to use the postgres testing framework.
+const DISABLE_POSTGRES_DOCKER_API = false
+
+// Config variables for spawning the postgres container.
+const CONTAINER_NAME = "test_postgresql"
+const LOCAL_PORT = "5433"
+const REMOTE_PORT = "5432"
+const POSTGRES_USER = "postgres"
+const POSTGRES_PASSWORD = "postgres"
+const POSTGRES_DB = "test_postgres"
+
+// Configs for connecting to postgres.
+const MAX_CONNECTION_ATTEMPTS = 20
+const ATTEMPT_TIMEOUT = 3 * time.Second
+
+// get a random temporary directory.
+func getDirectory(t *testing.T) string {
+	require := require.New(t)
+	dbDir, err := ioutil.TempDir("", "badgerdb")
+	if err != nil {
+		require.NoError(err)
+	}
+	return dbDir
+}
+
+func RunTestWithBadgerAndPostgresOptimized(t *testing.T, testFunction func(t *testing.T, postgres *Postgres)) {
+	var postgres *Postgres
+	postgresChannel := make(chan struct{})
+	go func() {
+		postgres = InitializeTestPostgresInstance(t)
+		postgresChannel <- struct{}{}
+	}()
+	testFunction(t, nil)
+	<-postgresChannel
+	testFunction(t, postgres)
```

Comment:
IMO running badger + postgres in parallel like this is of limited value. More useful IMO is to run badger tests in a single CI runner and postgres tests in a parallel runner. I.e. you run the entire suite (instead of each test) over badger or postgres in parallel.